spinn_front_end_common/interface/buffer_management/buffer_manager.py

```diff
@@ -177,6 +177,7 @@ def load_initial_buffers(self):
             total_data, "Loading buffers ({} bytes)".format(total_data))
         for vertex in self._sender_vertices:
             for region in vertex.get_regions():
+                vertex.rewind(region)
```

Comment:
I would say that it's not clear (certainly not from the PyNN documentation) 
whether multiple instances of run correspond to the expectation of "start from 
the beginning" or "resume from the previous position". I would imagine that 
there are use cases for both. Jamie here has indicated that most people tend to 
use it in a pause-resume mode but it seems to me that there's a case for support 
for both.

Reset is somewhat different from "rewind" because *that* means reinitialise the 
simulation altogether. Whereas for example rewind might be appropriate in a 
learning experiment, with a fixed input dataset, where you want to present the 
dataset back to the simulation after having run through it once already and 
learned new weight values. That's not the only use case for a rewind 
functionality, so it's not the case that one can use a workaround solution that 
only addresses the learning case, but I expect that this would be one of the 
most common scenarios.

On 30/10/15 09:10, Alan Stokes wrote:
> In spinn_front_end_common/interface/buffer_management/buffer_manager.py
> <https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/26#discussion_r43483143>:
>
>> @@ -177,6 +177,7 @@ def load_initial_buffers(self):
>>              total_data, "Loading buffers ({} bytes)".format(total_data))
>>          for vertex in self._sender_vertices:
>>              for region in vertex.get_regions():
>> +                vertex.rewind(region)
>
> This worries me a lot. Rewinding the vertex every single time forces the
> buffered manager to be a "reset, try again" mode. Whereas multiple runs is meant
> to be more of a pause and resume mode. Therefore Ive extracted the rewind here
> and made it a seperate method which is called rewind, and we will need to think
> how this is powered from the top level of PyNN, given that there is only reset
> and run as functions in the PyNN offical language.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/26/files#r43483143>.
>